docs/querying/api.md

```diff
@@ -1014,6 +1018,9 @@ Example:
 $ curl -X POST \
   -g 'http://localhost:9090/api/v1/admin/tsdb/delete_series?match[]=up&match[]=process_start_time_seconds{job="prometheus"}'
 ```
+
+NOTE: This endpoint deletes samples from series, but will not necessarily delete associated series metadata in every case (even after cleaning tombstones). Thus the series metadata API endpoints may still return metadata for series whose samples have been deleted for the selected time range. The exact extent of metadata deletion is an implementation detail that may change in the future.
```

Comment:
This is not correct, it does not delete samples from series. It puts in tombstones.

What it does is make it so the samples are no longer returned, and I'd keep the explanation to just that.

Reply:
> This is not correct, it does not delete samples from series. It puts in tombstones.

This is a bit too much language lawyering for me. The description of the whole endpoint description starts with "DeleteSeries deletes data for a selection of series in a time range.". So if we don't want to use the word "delete" for this kind of logical deletion (tombstoning), then we also named the endpoint incorrectly and would need to adjust the other descriptions as well.

> What it does is make it so the samples are no longer returned, and I'd keep the explanation to just that.

Happy to clarify that first part, but I think it's still important to point out that metadata may still be returned, because that really goes counter to people's expectations.